### PR TITLE
[#3649] feat(client): remove log4j impl from client, common and client-java-runtime

### PR DESCRIPTION
**Pull request by FANNG1**
### What changes were proposed in this pull request?

remove log4j impl from client-java-runtime

### Why are the changes needed?

Fix: #3649 

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
local spark env

### CODE REVIEW

**Review by mchades — APPROVED** [review]

File: clients/client-java/src/main/java/com/datastrato/gravitino/client/BaseSchemaCatalog.java [hunk 1]
```diff
@@ -194,7 +191,6 @@ public boolean dropSchema(NameIdentifier ident, boolean cascade) throws NonEmpty
     } catch (NonEmptySchemaException e) {
       throw e;
     } catch (Exception e) {
-      LOG.warn("Failed to drop schema {}", ident, e);
```
Comment (diqiu50) [inline clients/client-java/src/main/java/com/datastrato/gravitino/client/BaseSchemaCatalog.java:197]: After remove the log, we can't know why it failed.

Comment (jerryshao) [reply clients/client-java/src/main/java/com/datastrato/gravitino/client/BaseSchemaCatalog.java:197]: This is the issue in branch-0.5 that the semantic is not aligned.

Comment (mchades) [reply clients/client-java/src/main/java/com/datastrato/gravitino/client/BaseSchemaCatalog.java:197]: I think we can leave it as it is, We had better not change its original behavior in 0.5.1

Comment (mchades) [reply clients/client-java/src/main/java/com/datastrato/gravitino/client/BaseSchemaCatalog.java:197]: BTW, this will be fixed in 0.6.0

**Pull request merged**